assets/js/pokeapi.js

```diff
@@ -0,0 +1,32 @@
+const pokeApi = {}
+
+function convertPokeApiDetailToPokemon(pokeDetail) {
+    const pokemon = new Pokemon()
+    pokemon.number = pokeDetail.order
+    pokemon.name = pokeDetail.name
+
+    const types = pokeDetail.types.map(typeSlot => typeSlot.type.name)
+    const [type] = types
+    pokemon.types = types
+    pokemon.type = type
+
+    pokemon.image = pokeDetail.sprites.other.dream_world.front_default
+
+    return pokemon
+}
+
+pokeApi.getPokemonDetail = (pokemon) => {

```

Comment:
We need some error control in those promises using the method `Promise.catch()`

Reply:
you can add it after a then() like
```
return fetch(pokemon.url)
            .then(response => response.json())
            .then(convertPokeApiDetailToPokemon)
            .catch(error => console.error(error))
```

Reply:
Thank you for this reminder! Will do it.